.tree-sitter-lint/tree-sitter-lint-local/src/lib.rs

```diff
@@ -0,0 +1,31 @@
+use std::{path::Path, sync::Arc};
+
+use tree_sitter_lint::{
+    clap::Parser, tree_sitter::Tree, tree_sitter_grep::RopeOrSlice, Args, Config, Plugin, Rule,
+    ViolationWithContext,
+};
+
+pub fn run_and_output() {
+    tree_sitter_lint::run_and_output(args_to_config(Args::parse()));
+}
+
+pub fn run_for_slice<'a>(
+    file_contents: impl Into<RopeOrSlice<'a>>,
+    tree: Option<&Tree>,
+    path: impl AsRef<Path>,
+    args: Args,
+) -> Vec<ViolationWithContext> {
+    tree_sitter_lint::run_for_slice(file_contents, tree, path, args_to_config(args))
+}
+
+fn args_to_config(args: Args) -> Config {
```

Comment:
Both entry points rely on this "loading-of-the-config-file + specifying-the-plugins-and-rules-that-are-hard-coded-in-this-custom-local-binary to get a `Config`" so encapsulate that